src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Serialization/ExtendableClientCapabilitiesJsonConverter.cs

```diff
@@ -23,8 +25,45 @@ public override bool CanConvert(Type objectType)
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            var obj = JObject.Load(reader);
-            return obj.ToObject<ExtendableClientCapabilities>();
```

Comment:
Why'd this break VSCode?

Reply:
We weren't passing in the `serializer`. 

This is particularly consequential for VSCode because O# wraps the individual capabilities in a [`Supports`](https://github.com/OmniSharp/csharp-language-server-protocol/blob/6371386ee90dde8a1c419f3dc8fc94e0c1bc35f0/src/Protocol/Serialization/Converters/SupportsConverter.cs) capability, which was preventing appropriate deserialization.

Can't just pass in the `serializer` to the original `obj.ToObject<ExtendableClientCapabilities>()` as it would recursively call this `ReadJson` over and over again as `serializer` contains this converter..

Reply:
Ahhh ok, thanks for clarifying